js/pricing.js

```diff
@@ -101,15 +102,29 @@ $(
         /**
          * Submits consent transaction.
          *
-         * <p>Prepares Privacy, Support Agreement consents data and transaction Url for sending and
-         * calls send transaction function.
+         * <p>Prepares transaction Url, Privacy and Support Agreement consents data, generates
+         * the reCaptcha token and adds it to the request data. After that calls send transaction
+         * function.
          */
         function submitOrder() {
+            const transactionUrl = `${apiUrl}/transaction`;
             const privacyConsent = $privacyConsent.prop("checked");
             const supportAgreementConsent = $supportAgreementConsent.prop("checked");
-            const data = JSON.stringify({privacyConsent, supportAgreementConsent});
-            const transactionUrl = `${apiUrl}/transaction`;
-            sendPaymentTransaction(transactionUrl, data);
+
+            showRedirect();
+
+            grecaptcha.ready(() => {
+                grecaptcha
+                    .execute(reCaptchaSiteKey, {action: 'submitPaymentTransaction'})
+                    .then(function (token) {
+                        let data = {
+                            consent: {privacyConsent, supportAgreementConsent}
+                        };
+                        data.token = token;
+                        data = JSON.stringify(data);
```

Comment:
let's have 2 constants instead of having `let` and overriding it.
Also, please do not use `data` as a variable name.